test/models/ride_test.dart

```diff
@@ -0,0 +1,468 @@
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 6)),
+                end: DateTime.now().add(const Duration(hours: 8)),
+              ),
+              2),
+          false);
+    });
+    test('ride overlaps with time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.approved,
+                  startTime: DateTime.now().add(const Duration(hours: 7)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 6)),
+                end: DateTime.now().add(const Duration(hours: 8)),
+              ),
+              2),
+          true);
+    });
+  });
+  group('Ride.shouldShowInListView', () {
+    test('show in Listview', () {
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime.now().add(const Duration(hours: 2)),
+          endTime: DateTime.now().add(const Duration(hours: 4)),
+          status: RideStatus.pending);
+      expect(ride.shouldShowInListView(past: false), true);
+    });
+    test('show in past Listview', () {
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime.now().subtract(const Duration(hours: 4)),
+          endTime: DateTime.now().subtract(const Duration(hours: 2)),
+          status: RideStatus.approved);
+      expect(ride.shouldShowInListView(past: true), true);
+    });
+    test('does not show in past Listview', () {
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime.now().subtract(const Duration(hours: 4)),
+          endTime: DateTime.now().subtract(const Duration(hours: 2)),
+          status: RideStatus.withdrawnByRider);
+      expect(ride.shouldShowInListView(past: true), false);
+    });
+  });
+  group('Ride.equals', () {
+    final Ride ride0 = RideFactory().generateFake(
+      id: 1,
+      createdAt: DateTime(2022, 10),
+      start: 'Berlin',
+      startPosition: Position(1, 1),
+      startTime: DateTime(2022, 11),
+      end: 'Frankfurt',
+      endPosition: Position(2, 2),
+      endTime: DateTime(2022, 12),
+      seats: 1,
+      createDependencies: false,
+      driveId: 2,
+      riderId: 3,
+      status: RideStatus.approved,
+      price: NullableParameter(6),
+    );
+    test('same ride', () async {
+      expect(ride0.equals(ride0), true);
+    });
+    test('handle drive', () async {
+      final Drive drive = DriveFactory().generateFake(createDependencies: false);
+      expect(ride0.equals(drive), false);
+    });
+    test('different status', () async {
```

Comment:
Here you're testing this for a few random parameters again. I would suggest testing it for the ID and maybe one additional parameter, (or, if you want to test every parameter, do that and do it in a loop, please)